src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt

```diff
@@ -1,10 +1,77 @@
 package fr.acinq.lightning.utils
 
+import fr.acinq.lightning.channel.*
+import fr.acinq.lightning.db.OutgoingPayment
+import fr.acinq.lightning.io.SendPayment
+import fr.acinq.lightning.payment.PaymentPart
 import org.kodein.log.Logger
+import org.kodein.log.Logger.Level.*
 
 /**
  * This should be used more largely once https://kotlinlang.org/docs/whatsnew1620.html#prototype-of-context-receivers-for-kotlin-jvm is stable
  */
 interface LoggingContext {
-    val logger: Logger
+    val logger: MDCLogger
+}
+
+/**
+ * A simpler wrapper on top of [Logger] with better MDC support.
+ */
+data class MDCLogger(val logger: Logger, val staticMdc: Map<String, Any> = emptyMap()) {
+    inline fun debug(mdc: Map<String, Any> = emptyMap(), msgCreator: () -> String) {
+        logger.log(level = DEBUG, msgCreator = msgCreator, meta = staticMdc + mdc)
+    }
+
+    inline fun info(mdc: Map<String, Any> = emptyMap(), msgCreator: () -> String) {
+        logger.log(level = INFO, msgCreator = msgCreator, meta = staticMdc + mdc)
+    }
+
+    inline fun warning(ex: Throwable? = null, mdc: Map<String, Any> = emptyMap(), msgCreator: () -> String) {
+        logger.log(level = WARNING, error = ex, msgCreator = msgCreator, meta = staticMdc + mdc)
+    }
+
+    inline fun error(ex: Throwable? = null, mdc: Map<String, Any> = emptyMap(), msgCreator: () -> String) {
+        logger.log(level = ERROR, error = ex, msgCreator = msgCreator, meta = staticMdc + mdc)
+    }
+}
+
+/**
+ * Utility functions to build MDC for various objects without polluting main classes
+ */
```

Comment:
Not sure if grouping the `mdc()` classes here is better, it's a separate commit that I can drop.

Reply:
I think there are advantages and drawbacks to each approach and no clear winner, keeping them all here makes sense to me.